security/sanitizers.rst

```diff
@@ -0,0 +1,352 @@
+
+For reliable results, **always** rebuild your entire dependency tree with the same sanitizer configuration.
+
+Enabling Sanitizers
+-------------------
+
+Conan cannot infer sanitizer flags from settings automatically.
+You have to pass the appropriate compiler and linker flags (e.g., ``-fsanitize=`` or ``/fsanitize=address``) via profiles or toolchains.
+Conan toolchains (e.g., ``CMakeToolchain``, ``MesonToolchain``) will propagate flags defined in ``[conf]`` sections.
+
+Modeling and applying sanitizers using settings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to model sanitizer options so that the package ID is affected by them, you can
+:ref:`customize new compiler sub-settings <reference_config_files_customizing_settings>`. You should not need
+to modify ``settings.yml`` directly; instead add :ref:`the settings_user.yml <examples_config_files_settings_user>`.
+
+This approach is preferred because enabling a sanitizer alters the package ID, allowing you to build and use
+the same binary package with or without sanitizers. This is ideal for development and debugging workflows.
+
+Configuring sanitizers as part of settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you typically use a specific set of sanitizers or combinations for your builds, you can specify
+a sub-setting as a list of values in your ``settings_user.yml``. For example, for Clang:
+
+.. code-block:: yaml
+   :caption: settings_user.yml
+   :emphasize-lines: 3
+
+   compiler:
+     clang:
+       sanitizer: [null, Address, Leak, Thread, Memory, UndefinedBehavior, HardwareAssistanceAddress, KernelAddress, AddressUndefinedBehavior, ThreadUndefinedBehavior]
+
+This example defines a few common sanitizers. You can add any sanitizer your compiler supports.
+The ``null`` value represents a build without sanitizers. The above models the use of ``-fsanitize=address``,
+``-fsanitize=thread``, ``-fsanitize=memory``, ``-fsanitize=leak``, ``-fsanitize=undefined``, ``-fsanitize=hwaddress``,
+``-fsanitize=kernel-address``, as well as combinations like ``-fsanitize=address,undefined`` and ``-fsanitize=thread,undefined``.
+
+As the ``sanitizer`` setting is a list, it can be choose by one single value at time.
+As an workaround to support mutiple sanitizers at same time, you can define combinations like
+``AddressUndefinedBehavior`` and ``ThreadUndefinedBehavior``, as listed above.
+There is no limitation on the number of combinations you can define, but keep in mind that these are only tags
+to help you manage your builds. You still need to pass the appropriate flags to the compiler and linker accordingly.
+
+Adding sanitizers as part of the profile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another option is to add the sanitizer values as part of a profile. This way, you can easily switch between
+different configurations by using dedicated profiles.
+
+.. code-block:: ini
+   :caption: compiler_sanitizers/profiles/asan
+
+   include(default)
```

Comment:
This can be a bit confusing/not reproducible? Is this a gcc based or a clang profile?
Because the ``settings_user.yml`` above only added the sanitizer model to ``clang``, not ``gcc``, but the default profile in Linux systems is generally ``gcc``.

Maybe it is better to fully explicitly define the profile? Or if not, maybe a comment saying the default profile is a clang one?

Reply:
Fixed by 0779b86452f9fd6e45bc5ed971e1d388633276a3